docs/docs/user-guide.md

```diff
@@ -4,115 +4,142 @@ sidebar_position: 2
+annotated with `@Nullable`.
+
+On the other hand, `nullToEmpty` promises to handle `null` arguments, so its
+parameter is annotated with `@Nullable` to indicate that `nullToEmpty(null)` is
+a valid method call. Its body considers the case where the argument is `null`
+and won't throw `NullPointerException`. It cannot return `null`, so its return
+type is annotated with `@NonNull`.
 
 ```java
-void doSomething(@Nullable String x) {
-  print(emptyToNull(x));
-  // OK: print accepts a @Nullable String
+void doSomething() {
+  // OK: nullToEmpty accepts null but won't return it
+  int length1 = nullToEmpty(null).length();
 
-  String z = emptyToNull(x).toString();
-  // Not OK: emptyToNull(x) can be null
+  // Not OK: emptyToNull doesn't accept null; also, it might return null!
+  int length2 = emptyToNull(null).length();
 }
 ```
 
-Tools could then use the `@Nullable` information to determine that the first use
-is safe but the second is not.
-
-As far as JSpecify is concerned, `String` and `@Nullable String` are *different*
-types. A variable of type `String` can reference any string object. A variable
-of type `@Nullable String` can too, but it can also be null. This means that
-`String` is a *subtype* of `@Nullable String`, in the same way that `Integer` is
-a subtype of `Number`. One way to look at this is that a subtype narrows the
-range of possibilities. A `Number` variable can be assigned from an `Integer`
-but it can also be assigned from a `Long`. Meanwhile an `Integer` variable can't
-be assigned from a `Number` (since that `Number` might be a `Long` or some other
-subtype). Likewise, a `@Nullable String` can be assigned from a `String` but a
-`String` can't be assigned from a `@Nullable String` (since that might be null).
+Tools can use the `@Nullable` and `@NonNull` annotations to warn users about
+calls that are unsafe.
+
+As far as JSpecify is concerned, `@NonNull String` and `@Nullable String` are
+*different types*. A variable of type `@NonNull String` can reference any
+`String` object. A variable of type `@Nullable String` can too, but it can also
+be `null`. This means that `@NonNull String` is a *subtype* of `@Nullable
+String`, in the same way that `Integer` is a subtype of `Number`. One way to
+look at this is that a subtype narrows the range of possible values. A `Number`
+variable can be assigned from an `Integer` but it can also be assigned from a
+`Long`. Meanwhile an `Integer` variable can't be assigned from a `Number` (since
+that `Number` might be a `Long` or some other subtype). Likewise, a `@Nullable
+String` can be assigned from a `@NonNull String` but a `@NonNull String` can't
+be assigned from a `@Nullable String` (since that might be `null`).
 
 ```java
 @NullMarked
 class Example {
   void useNullable(@Nullable String x) {...}
-  void useNonNull(String x) {...}
-  void example(@Nullable String nullable, String nonNull) {
+  void useNonNull(@NonNull String x) {...}
```

Comment:
The `@NonNull` usages here are especially strange in method enclosed by `@NullMarked`. Maybe we're trending toward just not speaking of `@NullMarked` at all yet \[edit: Yep, it looks that way from the rest of the file :)\], in which case I'd remove it here.

Reply:
Good catch.